libunwind/include/libunwind.h

```diff
@@ -115,6 +121,18 @@ extern int unw_set_reg(unw_cursor_t *, unw_regnum_t, unw_word_t) LIBUNWIND_AVAIL
 extern int unw_set_fpreg(unw_cursor_t *, unw_regnum_t, unw_fpreg_t)  LIBUNWIND_AVAIL;
 extern int unw_resume(unw_cursor_t *) LIBUNWIND_AVAIL;
 
+#ifdef LIBUNWIND_HAVE_MSAN
+// unw_getcontext is implemented in assembly so it is rather difficult to
+// mark the MSan shadow as initialized from within the function. Instead we
+// use a macro wrapper when compiling with MSan to avoid false-positives.
+#define unw_getcontext(context)                                                \
```

Comment:
Could we perhaps use an inline function instead of a macro?

Reply:
I don't think we can since I _believe_ that having an exported symbol called unw_getcontext is required. But I'm not familiar with what level of compatibility with GNU libunwind is required.

Reply:
I think what I had in mind was something like:

```
extern "C" int __unw_getcontext(unw_context_t *);

inline int unw_getcontext(unw_context_t *ctx) {
  // do stuff
  return __unw_getcontext(ctx);
}
```

I am having a bit of trouble tracking down how `unw_getcontext` is implemented right now, but IIUC it's only an alias to `__unw_getcontext`, is that right? If so, I think the `inline` approach here would work, I think?

Reply:
We have to be careful to call unw_getcontext first to avoid registers being clobbered, but yes that would work if we don't need unw_getcontext to be an exported function. Maybe `extern inline __attribute__((always_inline))` could work.

Reply:
I think I'd defer to the libunwind regulars. @MaskRay perhaps?